### PR TITLE
feat: Loading screen state

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

Shows the loading screen status in the debug panel. It will show `current stage`, and if stage is scene loading, `assets loaded `and `asset to load`

image

## How to test the changes?

1. Launch the explorer
2. Check the debug panel to see the loading status

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by anicalbano — COMMENTED** [review]
🟢 Windows testing worked as expected! Adding a comment while we wait for Mac to fully approve.

Checks included with this test for smoke verification:
✔️ Teleport (via command, map and GP carrousel)
✔️ Backpack functionalities
✔️ Performed emotes
✔️ Chat and interactions


https://github.com/user-attachments/assets/7d45f41d-16f6-461a-8f39-63f0c0937c78


https://github.com/user-attachments/assets/9d243adb-658a-40d5-981a-0305fdb5c13d

**Review by popuz — CHANGES_REQUESTED** [review]
ah, sorry, wrong PR

File: Explorer/Assets/Scripts/Global/Dynamic/TeleportOperations/CompleteLoadingStatus.cs
Comment (popuz) [inline Explorer/Assets/Scripts/Global/Dynamic/TeleportOperations/CompleteLoadingStatus.cs:14]: formatting is a bit off

File: Explorer/Assets/DCL/UserInAppInitializationFlow/LoadingStatus.cs
Comment (popuz) [inline Explorer/Assets/DCL/UserInAppInitializationFlow/LoadingStatus.cs:15]: following to our design, instead of having `AnalyticsCallback` we need to create a decorated loading status (`LoadingStatusAnalyticsDecorator`) that will subscribe probably in constructor and unsubscribe whenever `== LoadingStage.Completed`. So basically, logic for lines 73-74 and 81-85 should be extracted there.  Let's still use this decorated class even after analytics unsubscribing, just to keep things a bit easier.

**Review by popuz — CHANGES_REQUESTED** [review]
improving loading screen is always welcoming :) 
Mine main concern is reporting Analytics approach, that should be changed in accordance to our design

**Review by anicalbano — COMMENTED** [review]
🟢 Verified again on Windows and working as expected! Waiting for Mac for approval


https://github.com/user-attachments/assets/abc32025-8f34-43c7-84eb-cb40e09d3ef8

**Pull request merged**
